[ninja-kitware/lapack] New port and tool for Fortran

Neumann-A opened the pull request:
uses CMakeDetermineFortranCompiler to determine the fortran compiler, user has to setup the env.
more lightweight than PR #1098

@cbezault hope you like this approach more

Neumann-A committed:
added ninja release from kitware

Neumann-A committed:
added possibility to discover a Fortran compiler

Neumann-A committed:
[lapack-reference] new port

Neumann-A committed:
restore variables after usage

cbezault commented on ports/lapack-reference/portfile.cmake:
```diff
@@ -0,0 +1,63 @@
+# LAPACKE Builds LAPACKE
+# LAPACKE_WITH_TMG Build LAPACKE with tmglib routines
+
+SET(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
```
Why are you doing this?

Neumann-A replied:
Pure Fortran libraries seem to not install any header or includes files. Consumers need to define the corresponding library calls/symbols themselves. Only If I build the additional c parts (lapacke or cblas) header files get generated/installed. 
But this seems to be consistent behavior which I also observed while dealing with #6786 since openblas does also not export any of the low level ("fortran") blas functions.

cbezault commented on scripts/cmake/vcpkg_find_acquire_program.cmake:
```diff
@@ -119,6 +120,28 @@ function(vcpkg_find_acquire_program VAR)
     set(URL "http://7-zip.org/a/7z1604.msi")
     set(ARCHIVE "7z1604.msi")
     set(HASH 556f95f7566fe23704d136239e4cf5e2a26f939ab43b44145c91b70d031a088d553e5c21301f1242a2295dcde3143b356211f0108c68e65eef8572407618326d)
+   elseif(VAR MATCHES "NINJA_KITWARE")
```
Just move all of the below into the NINJA block. We're happy to make kitware's ninja the default ninja.

Neumann-A replied:
done

Neumann-A committed:
fixed missing ) and guard against spaces in paths

Neumann-A committed:
added VCPKG_Fortran_FLAGS to pass specific compiler flags via triplet.

Neumann-A committed:
rename ninja_kitware to ninja as requested

Neumann-A committed:
removed message leftover from debugging

Neumann-A committed:
fix all uppercase Fortran use.
Handling of Fortran should be similar to cmakes syntax
Simplified IFs by introducing a Boolean variable

cenit commented on ports/lapack-reference/portfile.cmake:
```diff
@@ -0,0 +1,63 @@
+
+vcpkg_get_fortran_compiler()
+
+vcpkg_find_acquire_program(NINJA_KITWARE)
```
Since now NINJA_KITWARE -> NINJA, apart from being unresolved it should not be necessary, amirite? ;)

Neumann-A replied:
NINJA_KITWARE still matches NINJA so it still resolved. Was not aware of the fact that vcpkg_configure_cmake puts ninja on the path. So i changed removed it.

Neumann-A committed:
rename ninja_kitware in ninja in portfile

Neumann-A committed:
restore path variable in portfile

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into ninja_kitware

Neumann-A committed:
vcpkg_configure_cmake puts ninja on the path

Neumann-A committed:
fix whitespace; remove comments

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into ninja_kitware

# Conflicts:
#	scripts/cmake/vcpkg_build_cmake.cmake

Neumann-A committed:
fix missing )

Neumann-A committed:
Merge remote-tracking branch 'origin/ninja_kitware' into ninja_kitware

Neumann-A committed:
minimal comment refactor

cbezault commented:
Please move to #6939.
